buildpack/telemetry/dynatrace.py

```diff
@@ -5,23 +5,148 @@
 """
 import logging
 import os
+import json
 from urllib.parse import urljoin
 
 INGEST_ENDPOINT = "/api/v2/metrics/ingest"
 
 
-def is_enabled():
+from buildpack import util
+# Environment variables for Dynatrace OneAgent
+# Only passed to the agent if set as environment variable
+default_env = {
+    # -- Environment variables for the integration
+    # "DT_PAAS_TOKEN": required, also used for telegraf integration
+    # "DT_SAAS_URL": required, also used for telegraf integration
+    "DT_TENANT": None,  # required for agent integration, dynatrace environment ID
+    "DT_TENANTTOKEN": None,  # optional, default value is get from manifest.json which is downloaded along with the agent installer
+    # -- Environment variables for orchestration
+    "DT_LOCALTOVIRTUALHOSTNAME": None,  # optional, default not set
+    "DT_APPLICATIONID": None,  # optional, default not set
+    "DT_NODE_ID": None,  # optional, default not set
+    "DT_CLUSTER_ID": None,  # optional, default not set
+    "DT_TAGS": None,  # optional tags e.g. MikesStuff easyTravel=Mike
+    "DT_CUSTOM_PROP": None,  # optional metadata e.g. Department=Acceptance Stage=Sprint
+    # -- Environment variables for troubleshooting
+    "DT_LOGSTREAM": "stdout",  # optional
+    "DT_LOGLEVELCON": None,  # Use this environment variable to define the console log level. Valid options are: NONE, SEVERE, and INFO.
+    "DT_AGENTACTIVE": None,  # Set to true or false to enable or disable OneAgent.
+    # -- Networking environment variables
+    "DT_NETWORK_ZONE": None,  # optional, Specifies to use a network zone. For more information
+    "DT_PROXY": None, # Optional, When using a proxy, use this environment variable to pass proxy credentials.
+}
```

Comment:
We are not exposing these directly. In the update_config methods, these are injected into the java process only if they're set as environment variables.

One other point, I couldn't find any proper document on these environment variables. They're mentioned on different pages basically. Not sure all of them are directly valid for our use. Need to test this aspect.

Reply:
If we are not sure if the vars are valid for the scenario, let's not add them to the java process, else it may introduce some unexpected behavior.

Let's keep the env vars to the minimal required ones(we can confirm with the Proximux envvids which ones were being used)

Reply:
Only kept the ones used before